core/lib/Cardano/Address/Script.hs

```diff
@@ -434,9 +497,126 @@ instance FromJSON Script where
                         "Found object with no known key 'any', 'all' or 'some'"
                     (      _,       _,      _)  -> fail
                         "Found multiple keys 'any', 'all' and/or 'some' at the same level"
-
             String{} ->
                 parseKey v
-
             _ ->
                 Json.typeMismatch "Object or String" v
+
+fromScriptJson
+    :: FromJSON (Script elem)
+    => (Value -> Parser (Script elem))
+    -> (Value -> Parser (Script elem))
+    -> Value
+    -> Parser (Script elem)
+fromScriptJson parseElem backtrack v =
+    asum
+        [ parseElem v
+        , parseAnyOf v
+        , parseAllOf v
+        , parseAtLeast v
+        , parseActiveFrom v
+        , parseActiveUntil v
+        ] <|> backtrack v
+
+parseAnyOf
+    :: FromJSON (Script elem)
+    => Value
+    -> Parser (Script elem)
+parseAnyOf = withObject "Script AnyOf" $ \o ->
+    RequireAnyOf <$> o .: "any"
+
+parseAllOf
+    :: FromJSON (Script elem)
+    => Value
+    -> Parser (Script elem)
+parseAllOf = withObject "Script AllOf" $ \o ->
+    RequireAllOf <$> o .: "all"
+
+parseAtLeast
+    :: FromJSON (Script elem)
+    => Value
+    -> Parser (Script elem)
+parseAtLeast = withObject "Script SomeOf" $ \o -> do
+    some <- o .: "some"
+    RequireSomeOf <$> some .: "at_least" <*> some .: "from"
+
+parseActiveFrom
+    :: Value
+    -> Parser (Script elem)
+parseActiveFrom = withObject "Script ActiveFrom" $ \o ->
+    ActiveFromSlot <$> o .: "active_from"
+
+parseActiveUntil
+    :: Value
+    -> Parser (Script elem)
+parseActiveUntil = withObject "Script ActiveUntil" $ \o ->
+    ActiveUntilSlot <$> o .: "active_until"
+
+instance ToJSON Cosigner where
+    toJSON (Cosigner ix) = object ["cosigner" .= toJSON ix]
```

Comment:
It might be nicer to serialize cosigners as strings like `cosigner#1`, `cosigner#2` etc.. for it then allows to serialize the cosigner map as an object too by defining a `ToJSONKey Cosigner` instance.

Reply:
good idea.done, see state as in https://github.com/input-output-hk/cardano-addresses/pull/99/commits/070e6bf1e09840eb7f88e782baed7eeee56dcabe